common/server_protocolhandler.cpp

```diff
@@ -373,6 +383,20 @@ void Server_ProtocolHandler::pingClockTimeout()
 
     if (timeRunning - lastDataReceived > server->getMaxPlayerInactivityTime())
         prepareDestroy();
+
+    if (QString::fromStdString(userInfo->privlevel()).toLower() == "none")
+        if ((timeRunning - lastActionReceived) == round(server->getIdleClientTimout() *.9)) {
```

Comment:
Sure, that would work; just ensure to reset the bool every time you reset lastActionReceived